fdbserver/masterserver.actor.cpp

```diff
@@ -1206,27 +1222,27 @@ ACTOR Future<Void> serveLiveCommittedVersion(Reference<MasterData> self) {
 					                      req.debugID.get().first(),
 					                      "MasterServer.serveLiveCommittedVersion.GetRawCommittedVersion");
 
-				if (self->liveCommittedVersion == invalidVersion) {
-					self->liveCommittedVersion = self->recoveryTransactionVersion;
+				if (self->liveCommittedVersion.get() == invalidVersion) {
+					self->liveCommittedVersion.set(self->recoveryTransactionVersion);
 				}
 				++self->getLiveCommittedVersionRequests;
 				GetRawCommittedVersionReply reply;
-				reply.version = self->liveCommittedVersion;
+				reply.version = self->liveCommittedVersion.get();
 				reply.locked = self->databaseLocked;
 				reply.metadataVersion = self->proxyMetadataVersion;
 				reply.minKnownCommittedVersion = self->minKnownCommittedVersion;
 				req.reply.send(reply);
 			}
 			when(ReportRawCommittedVersionRequest req =
 			         waitNext(self->myInterface.reportLiveCommittedVersion.getFuture())) {
-				self->minKnownCommittedVersion = std::max(self->minKnownCommittedVersion, req.minKnownCommittedVersion);
-				if (req.version > self->liveCommittedVersion) {
-					self->liveCommittedVersion = req.version;
-					self->databaseLocked = req.locked;
-					self->proxyMetadataVersion = req.metadataVersion;
+				if (req.prevVersion.present() && (self->liveCommittedVersion.get() != invalidVersion) &&
+				    (self->liveCommittedVersion.get() < req.prevVersion.get())) {
+					auto dummy = waitForPrev(self, req);
+					req.reply.send(Void());
+				} else {
+					updateLiveCommittedVersion(self, req);
+					req.reply.send(Void());
```

Comment:
This `updateLiveCommittedVersion` actor is not waited and will just be cancelled, which is probably not what you want. If you wait here, then it will block the `loop choose` loop, which prevents serving `GetRawCommittedVersionRequest` while waiting. Instead, the code should be something like:
```
self->addActor(updateLiveCommittedVersion(self, req));
```
And move `req.reply.send(Void());` to the body of `updateLiveCommittedVersion`. In this way, the actor is not cancelled, and we are not blocked here.

The same treatment should be applied to the `if` block as well.

Reply:
thanks!

Reply:
`updateLiveCommittedVersion` doesn't appear to be an actor, so I think the else block is ok as is. The other block that uses `waitForPrev` though should follow this advice.